sklearn/linear_model/base.py

```diff
@@ -273,6 +273,11 @@ def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_
         """
         if self.fit_intercept:
+            if isinstance(self.coef_, np.ndarray):
```

Comment:
It's because when fitting `np.float32` data, `X_offset`, `y_offset`, and `X_scale` are of `np.float64` if not explicitly setting their type here, which would in turn makes `enet` fitting `np.float32` data have `np.float64` `coef_` in the end.